src/Migrations.ts

```diff
@@ -0,0 +1,204 @@
+   */
+  async readMigrations (): Promise<MigrationData[]> {
+    const migrationFiles = await this.getMigrationFiles()
+    if (!migrationFiles.length) {
+      throw new Error(
+        `No migration files found in '${this.config.migrationsPath}'.`
+      )
+    }
+
+    return Promise.all(
+      migrationFiles.map(migration => this.readMigrationData(migration))
+    )
+  }
+
+  // TODO: propose to make method GetInstance of the Migrations class, which will be async and we will call
+  // `downMigrations` and `upNewMigrations` functions in the migrate method. And user can call these
+  // `downMigrations` and `upNewMigrations` functions when ever he wants it
+  /**
+   * Performs a database migration.
+   *
+   * Migrates database schema to the latest version. Filename option is present
+   * in the the MigrationData if `readMigrations` function executes or user can put filename
+   * for each migration in the config.migrations
+   */
+  async migrate (): Promise<void> {
+    const table = this.config.table
+    const migrations = this.config.migrations ?? (await this.readMigrations())
+
+    // Create a database table for migrations meta data if it doesn't exist
+    await this.db.run(`CREATE TABLE IF NOT EXISTS "${table}" (
+      id   INTEGER PRIMARY KEY,
+      name TEXT    NOT NULL,
+      up   TEXT    NOT NULL,
+      down TEXT    NOT NULL
+    )`)
+
+    // Get the list of already applied migrations
+    let dbMigrations = await this.db.all(
+      `SELECT id, name, up, down FROM "${table}" ORDER BY id ASC`
+    )
+
+    await this.downMigrations(dbMigrations, migrations)
+    await this.upNewMigrations(dbMigrations, migrations)
+  }
+
+  /**
+   * Apply pending migrations
+   *
+   * @param dbMigrations exists migrations in the `table`
+   * @param migrations IMigrate.MigrationData
+   */
+  private async upNewMigrations (
+    dbMigrations: any[],
+    migrations: readonly MigrationData[]
+  ): Promise<void> {
+    const lastMigrationId = dbMigrations.length
+      ? dbMigrations[dbMigrations.length - 1].id
+      : 0
+    for (const migration of migrations) {
+      if (migration.id > lastMigrationId) {
+        await this.db.run('BEGIN')
+        try {
+          await this.db.exec(migration.up)
+          await this.db.run(
+            `INSERT INTO "${this.config.table}" (id, name, up, down) VALUES (?, ?, ?, ?)`,
+            migration.id,
+            migration.name,
+            migration.up,
+            migration.down
+          )
+          await this.db.run('COMMIT')
+        } catch (err) {
+          await this.db.run('ROLLBACK')
+          throw err
+        }
+      }
+    }
+  }
+
+  /**
+   * Undo migrations that exist only in the database but not in files,
+   * also undo the last migration if the `force` option is enabled.
+   *
+   * @param dbMigrations exists migrations in the `table`
+   * @param migrations IMigrate.MigrationData
+   */
+  private async downMigrations (
+    dbMigrations: any[],
```

Comment:
rename this to `appliedMigrations: MigrationData[]` so it's easier to tell what this is for